mypy/semanal.py

```diff
@@ -3546,6 +3551,14 @@ def visit_file(self, file: MypyFile, fnam: str, mod_id: str, options: Options) -
                 else:
                     typ = UnionType([UnboundType('__builtins__.str'),
                                      UnboundType('__builtins__.unicode')])
+            elif name == '__path__':
+                # __path__ is only defined on __init__.py files
+                if os.path.basename(fnam).lower() not in ('__init__.py', '__init__.pyi'):
+                    continue
+                # typ = self.sem.named_type('builtins.list', [self.sem.named_type('builtins.str')])
+                # typ = UnboundType('__builtins__.list', args=[UnboundType('__builtins__.str')])
+                # self.visit_import(Import(ids=[('typing', None)]))
+                typ = UnboundType('typing.List', args=[UnboundType('__builtins__.str')])
```

Comment:
Got it. Looks like this will need a bit larger refactor to implement. For example, leave the type undefined during the first pass of semantic analysis, and only set it during the second pass using `named_type` to construct the types. Using `__builtins__` to refer to a type is no longer a good approach because `list` doesn't support indexing.